src/index.js

```diff
@@ -49,6 +58,9 @@ export default function onClickOutsideHOC(WrappedComponent, config) {
     constructor(props) {
       super(props);
       this._uid = uid();
+      this.document = isBrowser()
+        ? config && config.getDocument ? config.getDocument(this.getInstance()) : getDocument()
```

Comment:
I meant that you should just make a config argument likes this:
```js
(WrappedComponent, {
  getDocument = defaultGetDocument,
  excludeScrollbar = false,
  handleClickOutside,
} = {}) =>
```

and that references to `config` should be just removed from the code.

Reply:
ok. You want me to remove references to config from here as well?

https://github.com/Pomax/react-onclickoutside/blob/master/src/index.js#L102

And change config argument below to default?
````
export default function onClickOutsideHOC(WrappedComponent, config) {
````

Reply:
@Andarist It seems the latest master branch breaks caching `this.document` cos `this.getInstance()`  and ref is only available after componentDidMount. Where can we cache document?

Reply:
You are right, you need to move caching this to the `componentDidMount`

Reply:
Yes, but this test fails

````
it('should register only one click outside listener per instance')

// this.document is not defined
````

When calling `this.props.enableOnClickOutside()` inside componendDidMount

Reply:
Hm, thats a bummer, child's `componentDidMount` gets executed first, so we wont be able to cache it first - problem is that you want to retrieve appropriate document from the instance

I would have to dig deeper into the issue to assess what can be done. Any ideas? Ideally you could pass document in advance (when creating the HOC) instead of reading it from the instance